spec/controllers/api/lists_controller_spec.rb

```diff
@@ -90,4 +91,250 @@
+      end
+
+      it "returns http success" do
+        expect(response).to have_http_status(:success)
+      end
+
+      it "JSON body response contains expected lists attributes" do
+        expect(@json_response.keys).to match_array(["created_at", "description", "id", "name", "public", "updated_at", "user_id", "movies"])
+      end
+      
+      it "JSON body response contains expected list name" do
+        expect(@json_response['name']).to eq(@list.name)
+      end
+      
+      it "JSON body response contains expected list description" do
+        expect(@json_response['description']).to eq(@list.description)
+      end
+
+      it "JSON body response contains expected list id" do
+        expect(@json_response['id']).to eq(@list.id)
+      end
+
+      it "JSON body response contains expected list state" do
+        expect(@json_response['public']).to eq(@list.public)
+      end
+
+      it "JSON body response contains user id of expected list" do
+        expect(@json_response['user_id']).to eq(@user.id)
+      end
+    end
+    context "when invalid" do
+      context "when the user does not authenticate" do
+        before do
+          get :show, format: :json, params: { id: @user.id }
+          @json_response = JSON.parse(response.body)
+          @nil_token = { "errors" => "Nil JSON web token" }
+        end
+  
+        it "returns http unauthorized" do
+          expect(response).to have_http_status(:unauthorized)
+        end
+
+        it "returns an error if token is nil" do
+          expect(@json_response).to eq(@nil_token)
+        end
+      end
+    end
+    
+    context "when the list does not exist" do
+      before do
+        request.headers["AUTHORIZATION"] = "Bearer #{@token}"
+        get :show, format: :json, params: { id: "False id" }
+        @json_response = JSON.parse(response.body)
+      end
+      
+      it "returns http unauthorized" do
+        expect(response).to have_http_status(:not_found)
+      end
+
+      it "The user does not exist" do
+        expect(@json_response['message']).to eq('The list does not exist')
+      end
+    end
+
+    context "when the user's token does not match the user to display" do
+      before do
+        request.headers["AUTHORIZATION"] = "Bearer #{@token}"
+        get :show, format: :json, params: { id: @list1.id }
+        @json_response = JSON.parse(response.body)
+      end
+      it "returns http unauthorized" do
+        expect(response).to have_http_status(:unauthorized)
+      end
+      it "The user does not exist" do
+        expect(@json_response['message']).to eq("You cannot see other users' lists")
+      end
+    end
+  end
+  describe "POST #create" do
+    before do
+      @created_list = { "message" => "Created List", "data" => { "id" => 9, "name" => "Create list", "description" => "This is my list", "public" => false, "user_id" => 1, "created_at" => "2020-07-01T22:01:58.137Z", "updated_at" => "2020-07-01T22:01:58.137Z" }}
```

Comment:
There lines are hard to read, you should divide this into multiple lines to improve readability